libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/workmanager/FetchNotificationsWorker.kt

```diff
@@ -122,6 +132,22 @@ class FetchNotificationsWorker(
         }
     }
 
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+        val notificationChannelId = notificationChannels.getChannelIdForMessage(false)
+        val notification = NotificationCompat.Builder(context, notificationChannelId)
+            .setSmallIcon(CommonDrawables.ic_notification)
+            .setOngoing(true)
+            .setTicker(context.getString(CommonStrings.common_android_notification_sync_notifications_foreground_service_title))
+            .setContentTitle(context.getString(CommonStrings.common_android_notification_sync_notifications_foreground_service_title))
+            .build()
+        val notificationId = NotificationIdProvider.getForegroundServiceNotificationId(ForegroundServiceType.NOTIFICATION_FETCHING)
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            ForegroundInfo(notificationId, notification, ServiceInfo.FOREGROUND_SERVICE_TYPE_REMOTE_MESSAGING)
```

Comment:
It seems that using this foreground type requires the manifest permission `android.permission.FOREGROUND_SERVICE_REMOTE_MESSAGING` (see the doc). Since this is dead code maybe not a big deal, but we should check that if it is used it works.